controllers/yawol-cloud-controller/targetcontroller/service_controller.go

```diff
@@ -57,14 +57,9 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	className, ok := svc.Annotations[yawolv1beta1.ServiceClassName]
-	if !ok {
-		className = ""
-	}
-
 	infraDefaults := GetMergedInfrastructureDetails(r.InfrastructureDefaults, svc)
 
-	if className != r.ClassName {
+	if !helper.CheckLoadBalancerClasses(svc, r.ClassNames) {
```

Comment:
Can we add a predicate that filters if a Service is relevant for this controller?

Reply:
Same as above

Reply:
We should be able to filter for this as well in the predicate. This is not actually related to this PR, so let's keep it as is.